notoptions-mitigation.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Plugin Name: VIP NotOptions Mitigation
+ * Description: Detect invalid values in the notoptions cache.
+ * Author: Automattic
+ * License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+namespace Automattic\VIP;
+
+use Automattic\VIP\Utils\Alerts;
+
+if ( is_admin() || ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+	add_action( 'muplugins_loaded', __NAMESPACE__ . '\notoptions_mitigation', -999 );
+}
+
+function notoptions_mitigation() {
+	global $wpdb;
+	$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+	if ( ! is_array( $notoptions ) ) {
+		return;
+	}
+
+	// filter for any values not equal to (bool)true.
+	$not_trues = array_filter( $notoptions, function( $v ) {
+		return true !== $v;
+	} );
+
+	$total_invalid = count( $not_trues );
+
+	// if they exist, something's borked
+	if ( 1 <= $total_invalid ) {
+
+		// attempt repair
+		$flushed = wp_cache_delete( 'notoptions', 'options' );
```

Comment:
Not sure if this is complicating things, but would it be worthwhile to add a role reset and potentially, reconnecting Jetpack? Maybe triggered via a cron event after.

Reply:
This might be a v2 when we have more information about frequency etc.

Reply:
+1 for keeping the first iteration simple...we can expand it after.